src/app/modules/webhook/webhook.utils.ts

```diff
@@ -14,3 +20,30 @@ export const formatWebhookResponse = (
   headers: stringifySafe(response?.headers) ?? '',
   data: stringifySafe(response?.data) ?? '',
 })
+
+/**
+ * Computes epoch of next webhook attempt based on previous attempts.
+ * @param previousAttempts Array of epochs of previous attempts
+ * @returns ok(epoch of next attempt) if there are valid retries remaining
+ * @returns err(WebhookNoMoreRetriesError) if there are no more retries remaining
+ */
+export const getNextAttempt = (
+  previousAttempts: number[],
+): Result<number, WebhookNoMoreRetriesError> => {
+  if (previousAttempts.length >= RETRY_INTERVALS.length)
+    return err(new WebhookNoMoreRetriesError())
+  const interval = RETRY_INTERVALS[previousAttempts.length]
+  const nextAttemptWaitTimeSeconds = randomUniform(
+    interval.base - interval.jitter,
+    interval.base + interval.jitter,
+  )
+  return ok(Date.now() + nextAttemptWaitTimeSeconds * 1000)
+}
+
+/**
+ * Converts an epoch to a readable format.
+ * @param epoch
+ * @returns the epoch represented as a readable string
+ */
+export const prettifyEpoch = (epoch: number): string =>
+  moment(epoch).tz('Asia/Singapore').format('D MMM YYYY, h:mm:ssa z')
```

Comment:
offtopic: we should really have a timezone env var or something lol

Reply:
maybe the day we become FormSEA or FormASEAN

Reply:
on a more serious note, maybe we can hardcode a `const TZ='Asia/Singapore'` variable somewhere